### PR TITLE
Integration tests.

### PR DESCRIPTION
**Pull request by Frozen**

### CODE REVIEW

File: pkg/node/state_fsm/fsm_common.go [hunk 1]
```diff
@@ -11,25 +9,6 @@ import (
 	"go.uber.org/zap"
 )
 
-func sendPeers(fsm FSM, p Peer, peers peer_manager.PeerManager) (FSM, Async, error) {
```
Comment (alexeykiselev) [inline pkg/node/state_fsm/fsm_common.go:14]: No more sending peers at all?

File: pkg/node/state_fsm/default.go [hunk 2]
```diff
@@ -27,5 +27,6 @@ func (a DefaultImpl) PeerError(fsm FSM, p Peer, baseInfo BaseInfo, _ error) (FSM
 
 func (a DefaultImpl) NewPeer(fsm FSM, p Peer, info BaseInfo) (FSM, Async, error) {
 	err := info.peers.NewConnection(p)
+	info.Reschedule()
```
[the remark this answers is not in the document]
Comment (Frozen) [reply pkg/node/state_fsm/default.go:30]: Reschedule checks consensus internally

File: pkg/proto/transactions.go [hunk 3]
```diff
@@ -166,7 +170,7 @@ type Transaction interface {
 	// Check that all transaction fields are valid.
 	// This includes ranges checks, and sanity checks specific for each transaction type:
 	// for example, negative amounts for transfers.
-	Valid() (bool, error)
+	Validate() (Transaction, error)
```
[the remark this answers is not in the document]
Comment (Frozen) [reply pkg/proto/transactions.go:173]: Definitely it should not modify content, but it just allow to write less code.

File: pkg/node/state_fsm/fsm_common.go (hunk 1 above)
Comment (Frozen) [reply pkg/node/state_fsm/fsm_common.go:14]: it was unused fn

File: pkg/proto/transaction_validators.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package proto
+
+import "github.com/wavesplatform/gowaves/pkg/errs"
+
+func ValidatePositiveAmount(amount int64, of string) error {
+	if !(amount > 0) {
```
[the remark this answers is not in the document]
Comment (Frozen) [reply pkg/proto/transaction_validators.go:6]: V.cond(amount > 0, TxValidationError.NonPositiveAmount(amount, "waves")),

File: pkg/state/diff_applier_test.go [hunk 5]
```diff
@@ -61,7 +61,7 @@ func TestDiffApplierWithWaves(t *testing.T) {
 	}
 	err = to.applier.applyBalancesChanges(changes, true)
 	assert.Error(t, err, "applyBalancesChanges() did not fail with leasing change leading to negative balance")
-	// Valid leasing change.
+	// Validate leasing change.
```
Comment (zer0main) [inline pkg/state/diff_applier_test.go:64]: "Valid" is correct here.

**Pull request merged**